android/shared/src/main/kotlin/kmp/android/shared/navigation/Destination.kt

```diff
@@ -0,0 +1,149 @@
+package kmp.android.shared.navigation
+
+import android.net.Uri
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.runtime.Composable
+import androidx.navigation.NamedNavArgument
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavDeepLink
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.dialog
+import com.google.accompanist.navigation.material.ExperimentalMaterialNavigationApi
+import com.google.accompanist.navigation.material.bottomSheet
+
+/**
+ * Base class for navigation graph root class.
+ */
+interface RootDestination {
+    val route: String
+}
+
+/**
+ * Base class for any destination the app uses.
+ */
+abstract class Destination(parent: RootDestination?) {
+    /**
+     * The root id of the destination graph used in the route <root>/<destinationId>/<argument>...
+     * In case the parent is null, we do not need any prefix and use the destinationId by itself <destinationId>/<argument>...
+     */
+    private val root: String = parent?.route?.let { "$it/" } ?: ""
```

Comment:
@vojta-horanek this is another change I did to your navigation, I added an interface for root and here we get path prefix from the root, if it exists (I thought it might be better than to trust users to prepend root id in the destination id string|.

Reply:
This is definitely an improvement! Very nice